src/query/phrase_query/phrase_scorer.rs

```diff
@@ -451,15 +451,6 @@ impl<TPostings: Postings> PhraseScorer<TPostings> {
             .docset_mut_specialized(0)
             .positions(&mut self.left_positions);
 
-        if self.num_terms == 2 {
```

Comment:
Remove it so it goes the normal code path and does nothing except for prepare data for `current_spans`.